src/Shared/ServerInfrastructure/BufferExtensions.cs

```diff
@@ -129,47 +129,43 @@ internal static void WriteAscii(ref this BufferWriter<PipeWriter> buffer, string
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    internal static unsafe void WriteNumeric(ref this BufferWriter<PipeWriter> buffer, ulong number)
+    internal static void WriteNumeric(ref this BufferWriter<PipeWriter> buffer, ulong number)
     {
         const byte AsciiDigitStart = (byte)'0';
 
-        var span = buffer.Span;
-        var bytesLeftInBlock = span.Length;
+        var start = buffer.Span;
```

Comment:
why this rename?

Reply:
It was using the name `start` below and the name `span` did not convey much information, hence I found start somewhat better. Happy to rename or change it back.

Reply:
For me `start` is more associated with an index, rather than with a "buffer".
Maybe rename the argument to `bufferWriter` and use `buffer` here for the span?